api/v1alpha1/stack_type.go

```diff
@@ -4,122 +4,125 @@ import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 
 // +kubebuilder:object:root=true
 // +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
-// +kubebuilder:resource:path=tinkerbell,scope=Namespaced,categories=tinkerbell,singular=tinkerbell,shortName=tb
+// +kubebuilder:resource:path=stack,scope=Namespaced,categories=stack,singular=stack
 // +kubebuilder:storageversion
 
-// Tinkerbell represents the tinkerbell stack that is being deployed in the kubernetes where the operator is deployed.
+// Stack represents the tinkerbell stack that is being deployed in the kubernetes where the operator is deployed.
 // Tinkerbell operator watches for different resources such as deployment, services, serviceAccounts, etc. One of those
-// CRs is Tinkerbell which the operator will install the tink-stack based on its specs. Once the CR is deleted,
+// CRs is Stack which the operator will install the tink-stack based on its specs. Once the CR is deleted,
 // the operator will delete all tinkerbell resources.
-type Tinkerbell struct {
+type Stack struct {
 	metav1.TypeMeta   `json:",inline"`
 	metav1.ObjectMeta `json:"metadata,omitempty"`
 
 	// Spec describes the desired tinkerbell stack state.
-	Spec TinkerbellStackSpec `json:"spec,omitempty"`
+	Spec StackSpec `json:"spec"`
 }
 
-// TinkerbellStackSpec specifies details of an Tinkerbell setup.
-type TinkerbellStackSpec struct {
+// StackSpec specifies details of an Tinkerbell setup.
+type StackSpec struct {
 	// Version is the Tinkerbell CRD version.
 	Version string `json:"version"`
 
 	// DNSResolverIP is indicative of the resolver IP utilized for setting up the nginx server responsible for proxying
 	// to the Tinkerbell service and serving the Hook artifacts.
-	DNSResolverIP string `json:"dnsResolverIP"`
+	DNSResolverIP string `json:"dnsResolverIP,omitempty"`
 
 	// Registry is the registry to use for all images. If this field is set, all tink service deployment images
 	// will be prefixed with this value. For example if the value here was set to docker.io, then boots image will be
 	// docker.io/tinkerbell/boots.
-	Registry string `json:"registry"`
+	Registry string `json:"registry,omitempty"`
 
-	// ImagePullSecret the secret name containing the docker auth config which should exist in the same namespace where
+	// ImagePullSecrets the secret name containing the docker auth config which should exist in the same namespace where
 	// the operator is deployed(typically tinkerbell)
-	ImagePullSecret []string `json:"imagePullSecret"`
+	ImagePullSecrets []string `json:"ImagePullSecrets,omitempty"`
```

Comment:
```suggestion
	ImagePullSecrets []string `json:"imagePullSecrets,omitempty"`
```
It has to start with a small letter, otherwise it fails validation.